src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -102,8 +103,7 @@ cdef class Buffer:
         self._message_callback = None
 
     def __dealloc__(self):
-        if self._hb_buffer is not NULL:
-            hb_buffer_destroy(self._hb_buffer)
+        hb_buffer_destroy(self._hb_buffer)
```

Comment:
I assume that the NULL check is redundant because it's already checked in hb_buffer_destroy, right?

Reply:
It is redundant because `hb_buffer_create()` never returns `NULL` as well as the other HarfBuzz create functions, they return a singleton in case of failure that are safe to pass to other HarfBuzz functions.